dynamo-future-file-extension/extension/Extension.cs

```diff
@@ -0,0 +1,128 @@
+{
+    public class FutureFileExtension : IExtension, ILogSource
+    {
+        private static MethodInfo openxmlmethod;
+        private static MethodInfo openJsonFileMethod;
+
+        public event Action<ILogMessage> MessageLogged;
+
+        public string UniqueId => "277edc98-d327-404e-979c-7c6e6f6956b2";
+
+        public string Name => "FutureFileExtension_File_Compatability";
+
+        public void Dispose()
+        {
+
+        }
+
+        private static string ToEnumString<T>(T type)
+        {
+            var enumType = typeof(T);
+            var name = Enum.GetName(enumType, type);
+            var enumMemberAttribute = ((EnumMemberAttribute[])enumType.GetField(name).GetCustomAttributes(typeof(EnumMemberAttribute), true)).Single();
+            return enumMemberAttribute.Value;
+        }
+
+        public void Ready(ReadyParams sp)
+        {
+            //Debugger.Launch();
+            //if dynamo is newer than 2.11, don't monkey patch anything.
+            if (sp.StartupParams.DynamoVersion > new Version(2, 12, 0))
+            {
+                MessageLogged(LogMessage.Info($"This extension only patches dynamo versions older than 2.12, current version is {Assembly.GetCallingAssembly().GetName().Version}, returning."));
+                return;
+            }
+
+            DoPatching();
+        }
+
+        public void DoPatching()
+        {
+            var harmony = new Harmony("Dynamo.FurureFileExtension.Patch.1");
+
+            var originalOpenFileFromPath = AccessTools.Method(typeof(DynamoModel), nameof(DynamoModel.OpenFileFromPath));
+            var prefix = typeof(FutureFileExtension).GetMethod(nameof(Prefix), System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.Public);
+            openxmlmethod = typeof(DynamoModel).GetMethod("OpenXmlFileFromPath", System.Reflection.BindingFlags.NonPublic | BindingFlags.Instance);
+            openJsonFileMethod = typeof(DynamoModel).GetMethod("OpenJsonFileFromPath", System.Reflection.BindingFlags.NonPublic | BindingFlags.Instance);
+            if(originalOpenFileFromPath != null && prefix != null && openxmlmethod != null && openJsonFileMethod != null)
+            {
+                harmony.Patch(originalOpenFileFromPath, new HarmonyMethod(prefix));
+            }
+        }
+        public static bool Prefix(DynamoModel __instance, string filePath, bool forceManualExecutionMode = false)
+        {
+
+            __instance?.Logger?.Log($"openfilepath has been patched from{Assembly.GetExecutingAssembly().FullName }");
+            var knownInputTypeNames = Enum.GetValues(typeof(NodeInputTypes)).Cast<NodeInputTypes>().Select(x => ToEnumString(x));
+            XmlDocument xmlDoc;
+            Exception ex;
+            if (DynamoUtilities.PathHelper.isValidXML(filePath, out xmlDoc, out ex))
+            {
+                openxmlmethod.Invoke(__instance, new object[] { xmlDoc, filePath, forceManualExecutionMode });
+                return false;
```

Comment:
why not return true an let the original method get invoked ?
You would have to remove the `openxmlmethod.Invoke(__instance, new object[] { xmlDoc, filePath, forceManualExecutionMode });` call

Reply:
What is better about this approach? It seems harder to reason about to me.

Reply:
changes code in the original, already released version? The intention of this extension is to patch existing legacy versions, and it won't run at all if the version is greater than 2.12 ... I guess it's possible a hotfix could cause issues.

will think about it, one problem is that if we modify the json in this method, we'd have to write it to disk for the original method to use it - I want to avoid modifying files on open or needing to write to disk.

Reply:
I was suggesting that we return true (let the original method run) only if the xml is valid (so when no modifications are needed)
But seeing that there is only a small chance to modify pre 2.12 versions...then I am good with the current state

Reply:
ah, thanks for explaining.

Reply:
Also since we are already loading the xml into memory..it might be faster to call the `openxmlmethod` directly from here